how-to/troubleshoot/update-verify-logs.rst

```diff
@@ -1,47 +1,248 @@
-Analyze Update Verify Logs
-==========================
+################################
+Debugging Update Verify Failures
+################################
 
+Before doing this, you should familiarize yourself with `how update verify works <https://firefox-source-docs.mozilla.org/tools/update-verify/index.html>`__.
+
+
+*****
 When?
------
+*****
 
-When update verify tasks fail it is your responsibility as releaseduty
-to analyze them and determine whether or not any action needs to be
-taken for any differences found.
+When update verify tasks fail it is your responsibility as releaseduty to analyze them and determine whether or not any action needs to be taken for any differences found.
 
+You may also be responsible for looking over update verify logs if changes have been made or there are other reasons to believe that we need some manual verification of update verify logs even when things are green. (For example, if there were update verify issues in a previous release, we commonly want extra verification of the next release's update verify logs.)
+
+****
 How?
-----
+****
 
-Update verify tasks that have failed usually have a ``diff-summary.log``
-in their artifacts. This file shows you all of the differences found for
-each update tested. In the diffs, ``source`` is an older version Firefox
-that a MAR file from the current release has been applied to, and
-``target`` is the full installer for the current release.
+Download the logs locally
+=========================
+There are a heck of a lot of update verify logs in your typical release graph. It's often useful to download them and look at them locally instead of using the Taskcluster UI or raw log in the browser. This should help you do that:
 
-Here's an example of a very alarming difference:
+.. code-block:: bash
+
+    task_group_id=xxxxxxx
+    mkdir -p update-verify-logs
+    taskcluster group list -c $task_group_id --format-string "{{ .Status.TaskID }} {{ .Task.Metadata.Name }}" | grep ".* release-update-verify" | cut -d' ' -f1 | xargs -I {} sh -c "echo 'Getting log for {}'; taskcluster task log {} > update-verify-logs/{}.log"
```

Comment:
Re-reading this, might want to change that `-c` into a `-a`. `-c` will only get tasks that are marked as `completed`, not `failed` which might be confusing? Or add a comment about how this will only get green tasks logs.

If you change it to `-a` the command might block if some update-verify jobs are still running though (as the `taskcluster task log` part will tail the log.

Reply:
Oh, yeah, good call. I'll add a note about that, and only running it after tasks are completed.